cucoslib/workers/recommender.py

```diff
@@ -431,11 +456,134 @@ def find_relative_similarity(self, input_stack, input_stack_vectors, filtered_re
+
+        # Send the user_profile information to Janus model to get back the user_persona
+        # Which is then used as an input along with a few others to hit PGM 
+        # Which gives us the response with list of packages and that is later on passed to Gremlin to get the version information for those packages
+        
+        janus_response = self.call_janus_model(user_profile)
+        inputTaskForPGM = []
+
+        for result in results:
+            details = result['details'][0]
+            resolved = details['_resolved']
+            self.log.info(result)
+            newArr = [r['package'] for r in resolved]
+            jsonObject = {
+                'ecosystem': details['ecosystem'],
+                'user_persona': "1",
+                'package_list': newArr
+            }
+            self.log.info(jsonObject)
+            inputTaskForPGM.append(jsonObject)
+
+            # Call PGM and get the response
+            pgm_response = self.call_pgm(inputTaskForPGM)
+
+            # Iterate and call Gremlin query
+            recommendation = {
+                'analysis': {
+                    'companion_packages': [],
+                    'alternate_packages': []
+                }
+            }
+
+            for result in pgm_response:
+                ecosystem = result['ecosystem']
+                user_persona = result['user_persona']
+                companion_packages = result['companion_packages']
+
+                companion_out_arr = recommendation['analysis']['companion_packages']
+
+                get_response = GraphDB().get_version_information(companion_packages, ecosystem)
+
+                reco_hash = self.get_recommendations_hash(get_response)
+
+                for pkg, ver in reco_hash.items():
+                    companion_out_arr.append(ecosystem + ':' + pkg + ':' + ver)
+
+            return recommendation
+
+
+    def get_recommendations_hash(self, results):
+        dictionary = {}
+        highest_usage_hash = {}
+        recommendations = {}
+
+        for result in results:
+            package = result['pkg']
+            version_object = result['ver']
+            package_name = package['name'][0]
+            version = version_object['version'][0]
+            dep_count = version_object['dependents_count'][0]
+
+            if package_name not in dictionary:
+                dictionary[package_name] = {
+                    'version': version,
+                    'dependents_count': dep_count
+                }
+                recommendations[package_name] = version
+            
+            else:
+                if dictionary[package_name]['dependents_count'] < dep_count:
+                    dictionary[package_name]['version'] = version
+                    dictionary[package_name]['dependents_count'] = dep_count
+                    recommendations[package_name] = version
+        
+        return recommendations
+
+
+    def execute_old_flow(self, arguments=None):
```

Comment:
If the code is not used, please remove it. We don't want to maintain code that is dead.